### DISCUSSION TITLE
Unmarshal function for custom values?

### DISCUSSION THREAD

**Discussion by JamesColeman-LW** [spf13/viper#1607] [Q&A]
There are sometimes when I want to design a custom value, such as using an iota, and in doing so I want to build a custom unmarshal function. YAML has `MarshalYAML` and `UnmarshalYAML`, and JSON has `MarshalJSON` and `UnmarshalJSON` which allows me to customize how data in a format is decoded. Does viper have something similar?

**Comment by sagikazarmark** [comment]
It does: https://sagikazarmark.hu/blog/decoding-custom-formats-with-viper/
